### PR TITLE
Added text.js tests

### PR DESCRIPTION
**Pull request by Brett-Parker**
Test for entity text.js

Functions tested;

getRotation()
getHorizontalAlignment()
getVerticalAlignment()
getBoundingRect()
closestPoint()
extremes()
intersectPoints()

### CODE REVIEW

**Comment by Brett-Parker** [comment]
@dubstar-04 can you give this an eyeball before I add any other functions to the test.

**Comment by dubstar-04** [comment]
> @dubstar-04 can you give this an eyeball before I add any other functions to the test.

Looks good thanks.

**Comment by Brett-Parker** [comment]
@dubstar-04 this is can be merged now. I haven't done;

width()
draw()
dxf()
snaps()
within()
touched()

These are often interaction based so harder to test but will review in future.

File: test/entities/text.test.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  expect(text.getBoundingRect()).toStrictEqual(expectedResult);
+
+  // Test for defined points
+  const text2 = new Text();
+  const pt1 = new Point(10, 10);
+  const pt2 = new Point(20, 20);
+
+  text2.points = [pt1, pt2];
+
+  const expectedResult2 = {'height': 0, 'width': 0, 'x': 10, 'y': 10};
```
Comment (dubstar-04) [inline test/entities/text.test.js:153]: BoundingBox can't be calculated without a canvas context.

File: test/entities/text.test.js [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  // Test for closestPoint
+  const text = new Text();
+  const pt1 = new Point(10, 10);
+  const pt2 = new Point(20, 20);
+
+  const testPoint = new Point(15, 15);
+
+  text.points = [pt1, pt2];
+
+  const expectedResult = [{'type': 'Point', 'x': 10, 'y': 10}, 7.0710678118654755];
```
Comment (dubstar-04) [inline test/entities/text.test.js:168]: This is testing the objects string representation. It should test the objects property value. i.e. testPoint.x and testPoint.y

File: test/entities/text.test.js [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+  // Test for closestPoint
+  const text2 = new Text();
+  const pt3 = new Point(10, 10);
+  const pt4 = new Point(20, 20);
+
+  const testPoint2 = new Point(20, 15);
+
+  text2.points = [pt3, pt4];
+
+  const expectedResult2 = [{'type': 'Point', 'x': 10, 'y': 10}, 11.180339887498949];
```
Comment (dubstar-04) [inline test/entities/text.test.js:181]: This is testing the objects string representation.

File: test/entities/text.test.js [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  // Test for closestPoint
+  const text3 = new Text();
+  const pt5 = new Point(10, 10);
+  const pt6 = new Point(20, 20);
+
+  const testPoint3 = new Point(0, 0);
+
+  text3.points = [pt5, pt6];
+
+  const expectedResult3 = [{'type': 'Point', 'x': 10, 'y': 10}, 14.142135623730951];
```
Comment (dubstar-04) [inline test/entities/text.test.js:194]: This is testing the objects string representation.

File: test/entities/text.test.js [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+  expect(text3.closestPoint(testPoint3)).toEqual(expectedResult3);
+});
+
+test('Test Text.extremes', () => {
```
Comment (dubstar-04) [inline test/entities/text.test.js:199]: This can't be tested without a valid boundingbox. see earlier comment.

File: test/entities/text.test.js [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+  const pt4 = new Point(351, 222.1);
+
+  text3.points = [pt3, pt4];
+
+  const expectedResult3 = [128, 128, 110, 110];
+
+  expect(text3.extremes()).toStrictEqual(expectedResult3);
+});
+
+test('Test Text.intersectPoints', () => {
```
Comment (dubstar-04) [inline test/entities/text.test.js:230]: This can't be tested without a valid boundingbox. see earlier comment.

**Review by dubstar-04 — CHANGES_REQUESTED** [review]
The boundingbox of the Text entity currently can't be calculated without a valid canvas context. I'm aware of a practical way to solve this. 

An objects string representation may change causing these tests to fail even when the property of the object may be valid. Note the object string contains long floats in some cases and these may change for different architectures, so it's more better to test a fixed number of decimal places using .toBeCloseTo() https://jestjs.io/docs/expect#tobeclosetonumber-numdigits

**Pull request closed without merging**